File: client/handymen-ng-app/src/app/model/search-params.ts
export class SearchParams {
  public firstName: string;
  public lastName: string;
  public trades: string[];
  public avgRatingFrom: number;
  public avgRatingTo: number;
  public address: string;
  public handymen: [];

  constructor(firstName: string, lastName: string, trades: string[], avgRatingFrom: number, avgRatingTo: number, address: string, handymen: []) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.trades = trades;
    this.avgRatingFrom = avgRatingFrom;
    this.avgRatingTo = avgRatingTo;
    this.address = address;
    this.handymen = handymen;
  }
}

File: server/handymen-backend/Model/models/SearchParams.cs
using System.Collections.Generic;

namespace Model.models
{
    public class SearchParams
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public List<string> Trades { get; set; }
        public int AvgRatingFrom { get; set; }
        public int AvgRatingTo { get; set; }
        public string Address { get; set; }
    }
}